src/sources/docker_logs.rs

```diff
@@ -1346,11 +1371,52 @@ mod integration_tests {
         );
     }
 
+    #[tokio::test]
+    async fn exclude_containers() {
+        trace_init();
+
+        let will_be_read = "12";
+        let included0 = "vector_test_include_0";
+        let included1 = "vector_test_include_1";
+        let excluded0 = "vector_test_exclude_0";
+
+        let docker = docker().unwrap();
+
+        let out = source_with_config(DockerLogsConfig {
+            include_containers: Some(
+                &[included0, included1]
+                    .iter()
+                    .map(|&s| s.to_owned())
+                    .collect(),
+            ),
+            exclude_containers: Some(&[excluded0].iter().map(|&s| s.to_owned()).collect()),
```

Comment:
I think this test would pass even if `exclude_containers` wasn't specified given that `include_containers` will already filter out `vector_test_exclude_0`. Maybe do `include_containers: Some(vec!["vector_test"])`? This would match all 3 and then the `exclude_containers` would filter out `vector_test_exclude_0`.

Reply:
Makes perfect sense 👍🏼